src/foam/nanos/dig/exception/DAONotFoundException.js

```diff
@@ -7,22 +7,34 @@ foam.CLASS({
   package: 'foam.nanos.dig.exception',
   name: 'DAONotFoundException',
   extends: 'foam.nanos.dig.exception.DigErrorMessage',
+  javaGenerateDefaultConstructor: false,
 
   axioms: [
     {
       name: 'javaExtras',
       buildJavaClass: function(cls) {
         cls.extras.push(`
-          public DAONotFoundException(String message) {
-            super(message);
-          } 
+          protected DAONotFoundException() { }
+
+          public DAONotFoundException(String daoName) {
+            setDaoName(daoName);
+            setMessage(getTranslation());
+          }
         `
         );
       }
     }
   ],
 
   properties: [
+    {
+      name: 'exceptionMessage',
+      value: 'DAO not found: {{daoName}}'
+    },
+    {
```

Comment:
This is not required. Set daoName is the message when the exception is generated.